analyses/oncoprint-landscape/01-plot-oncoprint.R

```diff
@@ -0,0 +1,254 @@
+
+#### Set Up --------------------------------------------------------------------
+
+# Install maftools
+if (!("maftools" %in% installed.packages())) {
+  install.packages("BiocManager")
+  BiocManager::install("maftools")
+}
+library(maftools)
+
+# Define a color vector for plots 
+colores <- c("Missense_Mutation" = "#35978f", 
+            "Nonsense_Mutation" = "#000000",
+            "Frame_Shift_Del" = "#56B4E9", 
+            "Frame_Shift_Ins" = "#FFBBFF", 
+            "Splice_Site" = "#F0E442",
+            "Translation_Start_Site" = "#191970",
+            "Nonstop_Mutation" = "#545454",
+            "In_Frame_Del" = "#CAE1FF",
+            "In_Frame_Ins" = "#FFE4E1",
+            "Stop_Codon_Ins" = "#CC79A7",
+            "Start_Codon_Del" = "#56B4E9",
+            "Fusion" = "#7B68EE",
+            "Multi_Hit" = "#f46d43",
+            "Hom_Deletion" = "#313695",
+            "Hem_Deletion" = "#abd9e9",
+            "Amplification" = "#c51b7d",
+            "Loss" = "#0072B2",
+            "Gain" = "#D55E00",
+            "High_Level_Gain" = "#FF0000",
+            "Multi_Hit_Fusion" = "#CD96CD")
+
+#### Functions -----------------------------------------------------------------
+
+# Get `magrittr` pipe
+`%>%` <- dplyr::`%>%`
+
+plot_oncoplot <- function(maf_object, filename){
+  # Given a maf file and a filename, plot an oncoprint of the variants in the 
+  # dataset, save as a png file and display.
+  # Args:
+  #   maf_object: name or path to a maf object
+  #   filename: name to save the png file as 
+  # Return:
+  #   oncoprint: plot produced using the maftools `oncoplot` function
+  
+  # Plot and save the oncoprint 
+  png(
+    file.path(plots_dir, filename),
+    width = 65,
+    height = 30,
+    units = "cm",
+    res = 300
+  )
+  oncoplot(maf_object,
+           clinicalFeatures = c(
+             "broad_histology",
+             "short_histology",
+             "reported_gender",
+             "tumor_descriptor",
+             "molecular_subtype"
+           ),
+           genes = genes,
+           logColBar = TRUE,
+           sortByAnnotation = TRUE,
+           showTumorSampleBarcodes = TRUE,
+           removeNonMutated = FALSE,
+           annotationFontSize = 0.7,
+           SampleNamefontSize = 0.5,
+           fontSize = 0.7,
+           colors = colores
+  )
+  dev.off()
+}
+
+#### Directories and Files -----------------------------------------------------
+
+# Detect the ".git" folder -- this will in the project root directory.
+# Use this as the root directory to ensure proper execution, no matter where
+# it is called from.
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# Path to the data obtained via `bash download-data.sh`.
+data_dir <- file.path(root_dir, "data")
+
+# Path to output directory for plots produced
+plots_dir <- file.path(root_dir, "analyses", "oncoprint-landscape", "plots")
+
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+genes <- NULL
```

Comment:
You have the logic down below now: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/176/files#diff-f1245b275299730475f8b3d538c277f6R204 so I think this can come out.